lib/improver/weighted_blend.py

```diff
@@ -102,16 +294,52 @@ def process(self, cube, weights=None):
                        ', cube shape is '
                        '{0:s}'.format(cube.coord(self.coord).points.shape))
                 raise ValueError(msg)
-            weights_array = iris.util.broadcast_to_shape(np.array(weights),
-                                                         cube.shape,
-                                                         collapse_dim)
-        # Calculate the weighted average.
-        result = cube.collapsed(self.coord,
-                                iris.analysis.MEAN, weights=weights_array)
+
+        # If coord to blend over is a scalar_coord warn
+        # and return original cube.
+        coord_dim = cube.coord_dims(self.coord)
+        if not coord_dim:
+            msg = ('Trying to blend across a scalar coordinate with only one'
+                   'value. Returning original cube')
+            warnings.warn(msg)
+            result = cube
+
+        # Blend the cube across the coordinate
+        # Use percentile Aggregator if required
+        elif perc_coord is not None:
+            percentiles = np.array(perc_coord.points, dtype=float)
+            perc_dim, = cube.coord_dims(perc_coord.name())
+            # Set equal weights if none are provided
+            if weights is None:
+                num = len(cube.coord(self.coord).points)
+                weights = np.ones(num) / float(num)
+            # Set up aggregator
+            PERCENTILE_BLEND = (Aggregator('percentile_blend',
+                                PercentileBlendingAggregator.aggregate))
+
+            result = cube.collapsed(self.coord,
+                                    PERCENTILE_BLEND,
+                                    arr_percent=percentiles,
+                                    arr_weights=weights,
+                                    perc_dim=perc_dim)
+
+        # Else do a simple weighted average
+        else:
+            # Equal weights are used as default.
+            weights_array = None
+            # Else broadcast the weights to be used by the aggregator.
+            if weights is not None:
+                weights_array = iris.util.broadcast_to_shape(np.array(weights),
```

Comment:
I'm slightly concerned we are not treating the case where a list (rather than an array) is passed in as weights.  There are some unit tests to test what happens if a list is passed in, but I think we are doing something different if we are blending probabilities compared to percentiles. I think we should probably not be recasting the input as an array. What do you think @carolinejjones ?